Renamed isAt* funcs to at* in Parser and Parser.Lookahead

natikgadzhi opened the pull request:
**Motivation**

Closes #1976. This pull request is a step towards making the Parser API consistent, and renaming some of the boolean `isAt*` functions with `at*`.

**Changes**
- [x] `Lexer.Cursor.isAtStringInterpolationAnchor` — private, so I'll rename it and change any spots where it's invoked. Deprecating the existing function should not be necessary.
- [x] `Lexer.Cursor.isAtEscapedNewline` — same as above.
- [x] `Parser.Lookahead` has two fileprivate functions `isAtStartOfPostfixExprSuffix` and `isInBindingPatternPosition` — I think it's best to rename both of them so the code in that particular extension is consistent. It would be weird to have `atStartOfPostfixExprSuffix` and `isIn*`. 
- [x] `Parser.isAtPythonStyleInheritanceClause`
- [x] In `SwiftParser/Statements.swift` we have `Parser.Lookahead` extension with `isAtStartOfSwitchCase`, `isStartOfStatement`, `isStartOfConditionalSwitchCases`. 
   - These are not marked as private, so I _think_ we should keep the original version and deprecate it.
   - Should I rename `isStartOf*` as well for consistency?
- [x] `SwiftParser/Types.swift` has `Parser.Lookahead.isAtFunctionTypeArrow` — also not marked as private.

**TODO**
- [x] Decide if we need deprecated versions for non-private funcs
- [x] Tests pass
- [x] Formatting OK

ahoppen commented on Sources/SwiftParser/Expressions.swift:
```diff
@@ -2487,7 +2487,7 @@ extension Parser.Lookahead {
   // Helper function to see if we can parse member reference like suffixes
   // inside '#if'.
-  fileprivate mutating func isAtStartOfPostfixExprSuffix() -> Bool {
+  fileprivate mutating func atStartOfPostfixExprSuffix() -> Bool {
```
I would keep the functions in `Lexer.Cursor` in the `isAt` style. The main checking function of `Lexer.Cursor` also is `is(offset: Int = 0, at: CharcterByte)`, so that fits with the `isAt` style. The parser just uses `at(_: TokenSpec)`.

natikgadzhi replied:
Got it! One sec, I'll clean this up.

ahoppen reviewed (commented):
Looks great. Thank you.

> Decide if we need deprecated versions for non-private funcs

These functions aren’t public, so there’s no need for deprecation.

natikgadzhi commented:
@ahoppen, thank you for reviewing this! Cleaned up Lexer, so I think it should be good to go now.

ahoppen commented on Sources/SwiftParser/Lexer/Cursor.swift:
```diff
@@ -1856,7 +1856,7 @@ extension Lexer.Cursor {
   /// Returns `true` if the cursor is positioned at `\##(` with `delimiterLength`
   /// pound characters.
-  private func isAtStringInterpolationAnchor(delimiterLength: Int) -> Bool {
+  private func atStringInterpolationAnchor(delimiterLength: Int) -> Bool {
```
Oh, I put the previous comment on the wrong line. I meant that we should keep the functions in Cursor.swift. Everything else should be renamed. I’m sorry.

natikgadzhi replied:
Ah, mb, I should've read it more carefully too. Okay, so reverting this change, but renaming SwiftParser/Expressions.swift.

ahoppen reviewed (approved):
Looking good 👍🏽

ahoppen commented:
@swift-ci Please test

natikgadzhi commented:
@ahoppen cleaned it up. Parser and Parser.Lookahead use at* notation. Cursor is back to isAt* notation.

ahoppen commented:
@swift-ci Please test

natikgadzhi commented:
@ahoppen I see the formatting issues on the CI output, and I can clean them up. I'll build swift-format from main as you advised and work through them.

ahoppen commented:
Great. Thank you for fixing the formatting issue. Also: The macOS CI failure is unrelated to your changes.

natikgadzhi commented:
My local `swift-format` is still building, but I think I've fixed the only problem it found. Rebased over main to see if the macOS test is fixed there.

ahoppen commented:
OK, let’s try again. 

Also, just a note: We prefer to squash the commits of a PR because it just creates a nicer git history. I’ll put together some documentation with reasons for that in the upcoming days.

@swift-ci Please test

natikgadzhi commented:
@ahoppen, agreed re: squashing — I usually end up doing that on the GitHub side when merging, but I'm happy to `git rebase -i` and squash so that after a review there's a single commit per pull request!

I usually keep commits separate to retain the step by step history of a PR while we're still talking through it, but once the review concludes, I can absulutely squash them.

natikgadzhi commented:
There's `Parser.isContextualExpressionModifier` — should we rename it as well for consistency? @ahoppen

ahoppen commented:
@swift-ci Please test Windows

ahoppen commented:
> There's Parser.isContextualExpressionModifier — should we rename it as well for consistency? @ahoppen

Yes, I think that would be a good idea as well.

natikgadzhi commented:
Cleaned it up, and squashed into a single commit.

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please test Windows
